app/knights/knights_preparing.py

```diff
@@ -0,0 +1,23 @@
+def prepare_knights(knights_config: dict) -> dict:
+
+    knights = {}
+
+    for knight in knights_config.values():
+        hp = knight["hp"]
+        power = knight["power"] + knight["weapon"]["power"]
+        protection = 0
+
+        if knight["armour"]:
+            protection += sum(arm["protection"] for arm in knight["armour"])
+
+        if knight["potion"] is not None:
+            if "hp" in knight["potion"]["effect"]:
```

Comment:
try to use settatr and for loop

Reply:
I can't understand how to use settatr in this case because I don't use any classes in this file, I'm working here only with keys and values of dicts for creating simple dict with useful total values